Modules/Scripted/DICOMLib/DICOMWidgets.py

```diff
@@ -906,11 +847,81 @@ def onNodeAdded(caller, event, calldata):
 
     self.onLoadingFinished()
 
+  def warnUserIfLoadableWarningsAndProceed(self):
+    warningsInLoadableWithConfidence = 0.0
+    maximumConfidence = 0.0
+    for plugin in self.loadablesByPlugin:
+      for loadable in self.loadablesByPlugin[plugin]:
+        if loadable.warning != "":
+          logging.warning('Warning in DICOM plugin ' + plugin.loadType + ' when examining loadable ' + loadable.name +
+                          ': ' + loadable.warning)
+          if warningsInLoadableWithConfidence < loadable.confidence:
+            warningsInLoadableWithConfidence = loadable.confidence
+        if maximumConfidence < loadable.confidence:
+          maximumConfidence = loadable.confidence
+    if warningsInLoadableWithConfidence == maximumConfidence and not self.advancedView:
+      warning = "Warnings detected during load.  Examine data in Advanced mode for details.  Load anyway?"
+      if not slicer.util.confirmOkCancelDisplay(warning):
+        return False
+    return True
+
+  def getAllSelectedLoadables(self):
+    loadables = {}
+    for plugin in self.loadablesByPlugin:
+      for loadable in self.loadablesByPlugin[plugin]:
+        if loadable.selected:
+          loadables[loadable] = plugin
+    return loadables
+
   def onLoadingFinished(self):
     if not self.browserPersistent:
       self.close()
 
 
+class DICOMReferencesDialog(qt.QMessageBox):
+
+  WINDOW_TITLE = "Referenced datasets found"
+  WINDOW_TEXT = "The loaded DICOM objects contain references to other datasets you did not select for loading. Please " \
+                "confirm if you would like to load the following referenced datasets."
+
+  def __init__(self, parent, loadables):
+    super(DICOMReferencesDialog, self).__init__(parent)
+    self.loadables = loadables
+    self.checkboxes = dict()
+    self.setup()
+
+  def setup(self):
+    self._setBasicProperties()
```

Comment:
If that helps, you create the UI using designer and include it as resources.
For example, see https://github.com/Slicer/Slicer/blob/eefeac9286f48552e8418a11f412b2823a09b407/Modules/Scripted/ExtensionWizard/CMakeLists.txt#L18-L20 that is referencing a ui file loaded dynamically.

It is also possible to install it in location relative the python script and load it from there.